### ISSUE TITLE
set_current_dir fails silently when passed "C:"

### ISSUE THREAD

**Issue by brson** [rust-lang/rust#32978]
This does not print "C:"

```
fn main() {
    use std::env;
    env::set_current_dir("C:").unwrap();
    println!("{:?}", env::current_dir().unwrap());
}
```

**Comment by retep998** [comment]
That is because `C:` is not an absolute path, but rather a relative path. When using a drive relative path such as `C:` it joins your relative path with the current directory for that drive to get the new current directory. Note that win32 does not allow you to set the drive specific current directories using `SetCurrentDirectory`. When the drive relative path does not match the current drive it pulls in those directories from magic environment variables typically set by cmd. For more information see the drive relative section on http://googleprojectzero.blogspot.com/2016/02/the-definitive-guide-on-win32-to-nt.html

tl;dr This is how Windows does things, so probably not a bug.

**Comment by tbu-** [comment]
This doesn't seem to be a bug, and could be closed. (?)

**Comment by drbo** [comment]
Even better: don't use relative paths on Windows.  They are broken in multithreaded programs (since the working directory is an unprotected global variable) and are completely unsupported in Windows Store apps.

**Comment by alexcrichton** [comment]
Regardless it appears that libstd isn't hiding anywhere or anything like that, so closing as not a bug.
